ablateLibrary/flow/compressibleFlow.cpp

```diff
@@ -1,34 +1,260 @@
-    : Flow(meshIn, name, arguments, initialization, boundaryConditions, {}), eos(eosIn) {
-    // Setup the problem
-    CompressibleFlow_SetupDiscretization(flowData, &mesh->GetDomain()) >> checkError;
+static const char* compressibleFlowComponentNames[TOTAL_COMPRESSIBLE_FLOW_COMPONENTS + 1] = {"rho", "rhoE", "rhoU", "rhoV", "rhoW", "unknown"};
+static const char* compressibleAuxComponentNames[TOTAL_COMPRESSIBLE_AUX_COMPONENTS + 1] = {"T", "vel", "unknown"};
+
+static PetscErrorCode UpdateAuxTemperatureField(FlowData_CompressibleFlow flowParameters, PetscReal time, PetscInt dim, const PetscFVCellGeom* cellGeom, const PetscScalar* conservedValues,
+                                                PetscScalar* auxField) {
+    PetscFunctionBeginUser;
+    PetscReal density = conservedValues[RHO];
+    PetscReal totalEnergy = conservedValues[RHOE] / density;
+
+    PetscErrorCode ierr = flowParameters->computeTemperatureFunction(NULL, dim, density, totalEnergy, conservedValues + RHOU, &auxField[T], flowParameters->computeTemperatureContext);
+    CHKERRQ(ierr);
+
+    PetscFunctionReturn(0);
+}
+
+static PetscErrorCode UpdateAuxVelocityField(FlowData_CompressibleFlow flowData, PetscReal time, PetscInt dim, const PetscFVCellGeom* cellGeom, const PetscScalar* conservedValues,
+                                             PetscScalar* auxField) {
+    PetscFunctionBeginUser;
+    PetscReal density = conservedValues[RHO];
+
+    for (PetscInt d = 0; d < dim; d++) {
+        auxField[d] = conservedValues[RHOU + d] / density;
+    }
+
+    PetscFunctionReturn(0);
+}
+
+ablate::flow::CompressibleFlow::CompressibleFlow(std::string name, std::shared_ptr<mesh::Mesh> mesh, std::shared_ptr<eos::EOS> eosIn, std::shared_ptr<parameters::Parameters> parameters,
+                                                 std::shared_ptr<fluxDifferencer::FluxDifferencer> fluxDifferencerIn, std::shared_ptr<parameters::Parameters> options,
+                                                 std::vector<std::shared_ptr<mathFunctions::FieldSolution>> initialization,
+                                                 std::vector<std::shared_ptr<boundaryConditions::BoundaryCondition>> boundaryConditions,
+                                                 std::vector<std::shared_ptr<mathFunctions::FieldSolution>> exactSolutions)
+    : Flow(name, mesh, parameters, options, initialization, boundaryConditions, {}, exactSolutions),
+      eos(eosIn),
+      fluxDifferencer(fluxDifferencerIn == nullptr ? std::make_shared<fluxDifferencer::AusmFluxDifferencer>() : fluxDifferencerIn) {
+    // Create a compressibleFlowData
+    PetscNew(&compressibleFlowData);
+
+    // make sure that the dm works with fv
+    const PetscInt ghostCellDepth = 1;
+    DM& dm = this->dm->GetDomain();
+    {  // Make sure that the flow is setup distributed
+        DM dmDist;
+        DMSetBasicAdjacency(dm, PETSC_TRUE, PETSC_FALSE) >> checkError;
```

Comment:
Is C++ smart enough not to create an object for this stream error check?

Reply:
I believe so.  We create a single global instance of the PetscErrorChecker.  The >> operator takes a const reference to the global instance.  I added a constructor/destructor and it appears to only be called once.